File: csharp/QueryString.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace QueryFileToolFramework
{
    class Program
    {
        static List<string> QueryFile(string queryString, string nowFolder)
        {
            var fileList = Directory.GetFiles(nowFolder);
            var filteredList = fileList.Where(f => f.Contains(queryString)).ToList();
            var subFolder1 = Directory.GetDirectories(nowFolder);
            var subFolder2 = subFolder1
                .Where(f => (!f.Contains("System Volume Information")) &&
                (!f.Contains("RECYCLE.BIN"))).ToList();
            subFolder2.ForEach(f =>
            filteredList.AddRange(QueryFile(queryString, nowFolder = f))
            );
            return filteredList;
        }
        static void Main(string[] args)
        {
            if (args.Length != 0)
            {
                var startTime = DateTime.Now.Ticks;
                var result = QueryFile(args[0], Directory.GetCurrentDirectory());
                long durningTime = DateTime.Now.Ticks - startTime;
                double durningTimeToSec = durningTime / 1000_0000.0;
                result.ForEach(f => Console.WriteLine(f));
                Console.WriteLine();
                Console.WriteLine($"查找共找到{result.Count}条结果，用时{durningTimeToSec}秒");
                Console.ReadLine();
            }
        }
    }
}

File: Python/QueryString.py
import os
import sys
import time


def QueryFile(queryString, nowFolder="."):
    # 列出当前文件夹下的所有文件
    fileList = os.listdir(nowFolder)
    # 遍历当前文件下的文件，把文件名包含所需字符串的文件的加入列表
    filteredList = [os.path.join(nowFolder, x) for x in fileList if
                    (x.find(queryString) != -1) and (os.path.isfile(os.path.join(nowFolder, x)))]
    # 获取当前文件夹下的子文件夹
    subFolders = [x for x in fileList if (os.path.isdir(os.path.join(nowFolder, x)))]
    for subFolder in subFolders:
        # windows系统跳过没有权限的管理员文件夹
        if subFolder.find("System Volume Information") > -1 or subFolder.find("RECYCLE.BIN") > -1:
            continue
            # 遍历子文件夹，结果拼接在一起
        filteredList.extend(QueryFile(queryString, nowFolder=os.path.join(nowFolder, subFolder)))
    return filteredList


if __name__ == '__main__':
    # 获取第一个参数，查找
    if len(sys.argv) > 1:
        T1 = time.perf_counter()
        result = QueryFile(sys.argv[1])
        T2 = time.perf_counter()
        for i in result:
            print(i)
        print(f"查找{sys.argv[1]} 共{len(result)}条结果，耗时{T2 - T1}")
